test.rb

```diff
@@ -22,11 +22,44 @@
 require 'minitest/autorun'
 require 'minitest/around/unit'
 require 'active_record'
+require 'action_pack'
 
 if ActiveSupport::VERSION::MAJOR == 3
   require 'active_support/core_ext/logger'
 end
 
+if ActionPack::VERSION::MAJOR > 3
+  require 'action_controller'
+end
+
+# Handle an edge-case when using Arel 5 (i.e. Rails <= 4.1) with Ruby >= 2.4:
```

Comment:
Using Rails <= 4.1 with Ruby >= shouldn't really be supported, to be honest.

* Rails 5.0 was released in June 2016, thus ending all support (including severe security fixes) for Rails 4.1. 
* Ruby 2.4 was released _after_ that, in December.

With gems specifically used as plugins of Rails, I tend to follow those support guidelines as no one should assume a gem would guarantee support for a version spread like that when the framework itself definitely does not.

However, since you've already gone through a lot of in-depth trouble here to make this work, I'll certainly merge it. Please note if there are any issues with this patch it'll likely just get removed in the future, though.

Reply:
Sounds good! Yeah, I agree this edge-case shouldn't happen in reality and that's definitely ok to remove it in the future (I think I encountered this while testing the Rails/Ruby versions combinations manually).